pypeit/specobjs.py

```diff
@@ -212,6 +212,7 @@ def unpack_object(self, ret_flam=False, extract_type='OPT'):
         meta_spec['PYP_SPEC'] = self.header['PYP_SPEC']
         meta_spec['PYPELINE'] = self[0].PYPELINE
         meta_spec['DET'] = detector
+        meta_spec['DISPNAME'] = self.header['DISPNAME']
```

Comment:
is `DISPNAME` guaranteed to exist?  I forget.

Reply:
I believe so, bu it is very difficult to figure out. I am trying to figure this out, but the spec2dobj code is pretty difficult to follow. I cannot find where DISPNAME is being propagated through. 

My suggestion is that we make a set of header cards or an extension with meta data part of the spec2d data model, since right now we have a rigid data model for spec2d, but a completely fluid set of header cards and other meta data that can be written to the file.

Reply:
good suggestion.  But is this line of code going
to crash things as is?

Reply:
No it does not crash.